src/components/Tooltip/BaseGenericTooltip/index.native.tsx

```diff
@@ -0,0 +1,132 @@
+    const [wrapperMeasuredHeight, setWrapperMeasuredHeight] = useState<number>();
+    const textContentRef = useRef<RNText>(null);
+    const viewContentRef = useRef<RNView>(null);
+    const rootWrapper = useRef<RNView>(null);
+
+    const StyleUtils = useStyleUtils();
+
+    useEffect(() => {
+        if (!textContentRef.current && !viewContentRef.current) {
+            return;
+        }
+        textContentRef.current?.measure((x, y, width) => setContentMeasuredWidth(width));
+        viewContentRef.current?.measure((x, y, width) => setContentMeasuredWidth(width));
+    }, []);
+
+    const {animationStyle, rootWrapperStyle, textStyle, pointerWrapperStyle, pointerStyle} = useMemo(
+        () =>
+            StyleUtils.getTooltipStyles({
+                tooltip: rootWrapper.current,
+                currentSize: animation,
+                windowWidth,
+                xOffset,
+                yOffset,
+                tooltipTargetWidth: targetWidth,
+                tooltipTargetHeight: targetHeight,
+                maxWidth,
+                tooltipContentWidth: contentMeasuredWidth,
+                tooltipWrapperHeight: wrapperMeasuredHeight,
+                manualShiftHorizontal: shiftHorizontal,
+                manualShiftVertical: shiftVertical,
+                shouldForceRenderingBelow,
+                shouldForceRenderingLeft,
+                wrapperStyle,
+            }),
+        [
+            StyleUtils,
+            animation,
+            windowWidth,
+            xOffset,
+            yOffset,
+            targetWidth,
+            targetHeight,
+            maxWidth,
+            contentMeasuredWidth,
+            wrapperMeasuredHeight,
+            shiftHorizontal,
+            shiftVertical,
+            shouldForceRenderingBelow,
+            shouldForceRenderingLeft,
+            wrapperStyle,
+        ],
+    );
+
+    let content;
+    if (renderTooltipContent) {
+        content = <View ref={viewRef(viewContentRef)}>{renderTooltipContent()}</View>;
+    } else {
+        content = (
+            <Text
+                numberOfLines={numberOfLines}
+                style={textStyle}
+            >
+                <Text
+                    style={textStyle}
+                    ref={textRef(textContentRef)}
+                >
+                    {text}
+                </Text>
+            </Text>
+        );
+    }
+
+    return (
+        <Animated.View
+            ref={viewRef(rootWrapper)}
+            style={[rootWrapperStyle, animationStyle]}
+            onLayout={(e) => {
+                const {height} = e.nativeEvent.layout;
+                if (height === wrapperMeasuredHeight) {
+                    return;
+                }
+                setWrapperMeasuredHeight(height);
+            }}
+        >
+            {content}
+            <View style={pointerWrapperStyle}>
+                <View style={pointerStyle} />
+            </View>
+        </Animated.View>
+    );
```

Comment:
I must wrap my head around this... On the Web, we use `ReactDOM.createPortal`, and on Native we don't even try to do anything similar, we just return the equivalent `Animated.View` directly. Have you encountered any limitations of this approach during testing, or does it just appear to work?

Reply:
Great to hear that!